src/atoms/referencesState.ts

```diff
@@ -23,3 +29,39 @@ export const setReferencesAtom = atom(null, (_get, set, references: ReferenceIte
 });
 
 export const referencesSyncInProgressAtom = atom<boolean>(false);
+
+const removeReferenceAtom = atom(null, (get, set, id: string) => {
+  const references = get(getReferencesAtom);
+  const newReferences = references.filter((ref) => ref.id !== id);
+  set(setReferencesAtom, newReferences);
+});
+
+export const removeReferencesAtom = atom(null, async (get, set, ids: string[]) => {
+  const referencesToRemove = ids
+    .map((id) => get(getDerivedReferenceAtom(id)))
+    .filter((ref) => ref)
```

Comment:
I think this this line is unnecessary, as the next line (`.filter(isNonNullish)`) does the same